specutils/io/registers.py

```diff
@@ -167,3 +169,37 @@ def _load_user_io():
                     import_module(file[:-3])
                 except ModuleNotFoundError:  # noqa
                     pass
+
+
+def identify_spectrum_format(filename):
+    """ Attempt to identify a spectrum file format
+
+    Given a filename, attempts to identify a valid file format
+    from the list of registered specutils loaders.  Essentially a wrapper for
+    `~astropy.io.registry.identify_format` setting **origin** to ``read`` and
+    **data_class_required** to `~specutils.Spectrum1D`.
```

Comment:
I am curious if this can be generalised to other `spectra` objects, e.g. by adding a `dtype=Spectrum1D` kwarg.
There are one or two loaders for `SpectrumList` now, and #708 should add one for `SpectrumCollection`, so it would be great if the same function could be used for identification. Does not have to be solved in this PR, but then the documentation would also need to be updated a little bit.

Reply:
I made a small change to allow a user to specify a dtype that can either be `Spectrum1D`, `SpectrumList`, or `SpectrumCollection`, with the default being `Spectrum1D`.  It throws an assert otherwise.